spec/index.md

```diff
@@ -1676,6 +1690,16 @@ If this traps (e.g. runs out of cycles), the state changes from the `cleanup` fu
 
 There must be at most one call to `ic0.call_on_cleanup` between `ic0.call_new` and `ic0.call_perform`.
 
+-   `ic0.call_with_best_effort_response : (timeout_seconds : i32) -> ()`
+
+    Relaxes the response delivery guarantee to be best effort, asking the system to respond at the latest after `timeout_seconds` have elapsed. Best effort means the system may also respond with a `SYS_UNKNOWN` reject code, signifying that the call **may or may not** have been processed by the callee. Then, even if the callee produces a response, it will not be delivered to the caller.
+
+    Any value for `timeout_seconds` is permitted, but is silently bounded from above by the `MAX_CALL_TIMEOUT` system constant; i.e., larger timeouts are treated as equivalent to `MAX_CALL_TIMEOUT` and do not cause an error. The implementation may add a specific [error code](#error-codes) to a reject message to indicate the cause, in particular whether the timeout expired. Note that the reject callback may be executed (possibly significantly) later than the specified time (e.g., if the caller is under high load), or before timeout expiration (e.g., if the system is under load).
```

Comment:
The callee can learn the timeout via `ic0.msg_deadline`.

Reply:
I think Bjoern is referring to the `MAX_CALL_TIMEOUT`?
I agree that it's unfortunate, but it's no less unfortunate than all the other system constants, such as instruction limits, message size limits, (stable) memory size limits, the various costs etc etc. I think it doesn't make sense to do a one-off for this constant. But I'm not sure what a good general solution is. We have so far (for good reason, IMO) steered clear from returning structured data in API calls. Maybe with [WIT](https://component-model.bytecodealliance.org/design/wit.html) it's time to reconsider? Otherwise, having a `get_constant_X` system API call per constant is super yucky. I suppose we could have a single `get_constant(constant_name)`, but that would require a uniform return type.

Reply:
@luc-blaeser wants to design a system API returning parameters such as remaining instructions, wasm heap etc. Maybe this would be an opportunity to reveal `MAX_CALL_TIMEOUT`.

Reply:
Makes sense, let's include it in the limits.